README.md

```diff
@@ -4,36 +4,148 @@ Fast, cross-platform mail server.
+  MiB of binary size.
+* `mysql`
+  Include support for MySQL driver in go-sqlmail.
+* `postgresql`
+  Include support for PostgreSQL driver in go-sqlmail.
+
+## Configuration
+
+### Syntax
+
+Maddy uses configuration format similar (but not the same!) to Caddy's
+Caddyfile.  You may want to read [Caddyfile page from Caddy docs](https://caddyserver.com/docs/caddyfile).
+
+Notable differences from Caddy's format:
+* Directives can't span multiple lines
+* Environmental variables are not expanded
+* You can't omit braces if you have only one configuration block
 
+### Modularity
+
+Maddy does have a module-based design and you specify in configuration modules
+you want to use and how you want them to work.
+
+Generic syntax for configuration block is as follows:
 ```
-imaps://127.0.0.1:1993 {
-	proxy imaps://mail.example.org
-	tls self_signed
-	compress
-	pgp
+module_name instance_name {
+    configuration_directives
 }
+```
+You can omit braces if there is no configuration directives in block:
+```
+module_name instance_name
+```
+
+`instance_name` is the unique name of the configuration block. It is used when
+you need to refer to the module from different place in configuration (e.g.
+configure SMTP to deliver mail to certain specific storage)
+
+You can omit `instance_name`. If there is only one module config. block - it
+will get name the same as `module_name`. However, you can't define multiple
+config. blocks this way because names should be unique.
+
+### Global options
+
+Certain options can be specified outside of any configuration block. They
+specify defaults for all configuration blocks unless they override them. Below
+are options that can be used like that:
+
+* `hostname <domain>`
+  Specify local hostname for all modules. Relevant for SMTP endpoints and queues.
+
+* `tls <cert_file> <pkey_file>`
+  Default TLS certificate to use. See
+  [CONFIG_REFERENCE.md](CONFIG_REFERENCE.md) for details.
+
+### Defaults
+
+Maddy provides reasonable defaults so you can start using it without spending
+hours writing configuration files. All you need it so define smtp and imap
+modules in your configuration, configure TLS (see below) and set domain name.
+
+Here is the minimal example to get you started:
+```
+tls cert_file pkey_file
+hostname emersion.fr
+
+imap imap://0.0.0.0 imaps://0.0.0.0
+smtp smtp://0.0.0.0:25
+submission smtp://0.0.0.0:587 smtps://0.0.0.0:465
+```
+Don't forget to use actual values instead of placeholders.
+
+With this configuration, maddy will create an SQLite3 database for messages in
+current directory and use it to store all messages.
 
-smtps://127.0.0.1:1025 {
-	proxy smtps://mail.example.org
-	tls self_signed
-	pgp
+### go-sqlmail: Database location
+
+If you don't like SQLite3 or don't want to have it in the current directory,
+you can override the configuration of the default module.
+
+See [go-sqlmail repository](https://github.com/foxcpp/go-sqlmail) for
+information on RDBMS support.
+
+```
+sqlmail default {
```

Comment:
Maybe we can just name this "sql" since it's alreayd obvious we're working with e-mails

Reply:
I named it this way because this is how "subproject" is named (go-sqlmail) but guess I will rename it to go-imap-sql, so yeah, it should be just `sql` now.

Reply:
Now I renamed go-sqlmail to go-imap-sql and module is named just 'sql'.